### PR TITLE
feat: reply message #254

### PR DESCRIPTION
**Pull request by FrolYo**
This is a very first version of reply functionality.

Done:
-------------------------
- Displaying message that is currently being replied above message input
- Displaying quote (replied message) inside chat list
- Navigating to replied message in list on click from both chat list and message editor

TBD:
-------------------------
- Fix navigation to chat entry - especially if it's from different tile
- Fix dozen of UI defects
- Highlight replied message on navigation

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/NavigateToChatEntryEvent.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+namespace ActualChat.Chat.UI.Blazor.Components;
+
+public record NavigateToChatEntryEvent(long ChatEntryId);
```
Comment (evgeniy-ac) [inline src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/NavigateToChatEntryEvent.cs:3]: It's named like an event but seems like a command
https://jimmybogard.com/message-naming-conventions/#:~:text=Event%20naming%20conventions,Subject

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor [hunk 2]
```diff
@@ -10,91 +11,115 @@
...
 
     [Parameter, EditorRequired, ParameterComparer(typeof(ByValueParameterComparer))]
     public ChatMessageModel Model { get; set; } = null!;
 
     protected override async Task OnInitializedAsync() {
+        var chatId = Model.Entry.ChatId;
         PrincipalId = await ChatAuthor
-            .GetOwnPrincipalId(Session, Model.Entry.ChatId, CancellationToken.None)
+            .GetOwnPrincipalId(Session, chatId, CancellationToken.None)
             .ConfigureAwait(true);
```
Comment (evgeniy-ac) [inline src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor:112]: false?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/UI.Blazor/Events/EventAggregator.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            var eventHandlers = _handlers[typeof(TEvent)];
+            var updatedList = eventHandlers.Remove(handler);
+            if (updatedList.Count == 0)
+                _handlers.Remove(typeof(TEvent));
+            else
+                _handlers[typeof(TEvent)] = eventHandlers;
+        }
+    }
+
+    public async Task Publish<TEvent>(TEvent @event, CancellationToken cancellationToken = default) where TEvent : class
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/UI.Blazor/Events/EventAggregator.cs:25]: image

[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/UI.Blazor/Events/EventAggregator.cs:25]: So I'd leave it with default value as in interface

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/NavigateToChatEntryEvent.cs (hunk 1 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/NavigateToChatEntryEvent.cs:3]: removed suffix

File: src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ActiveReply.razor [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        var chatMessage = new ChatMessageModel(chatEntry, MarkupParser.Parse(chatEntry.Content), ImmutableArray<TextEntryAttachment>.Empty) {
+            IsBlockStart = true,
+            IsQuote = true,
+        };
+        return chatMessage;
+    }
+
+    private void OnCloseClick() {
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ActiveReply.razor:36]: done

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor (hunk 2 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor]: done

File: src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ChatMessageEditor.razor [hunk 5]
```diff
@@ -123,4 +134,9 @@
     private async Task OnUploadFilesRequested() {
         await JSRef.InvokeVoidAsync("showFilesPicker").ConfigureAwait(true);
     }
+
+    private async Task Focus(CancellationToken cancellationToken) {
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ChatMessageEditor.razor:138]: done

**Pull request merged**